.gitignore

```diff
@@ -7,13 +7,10 @@ _ReSharper*/
 
 # Build output
 build/
-artifacts/
-src/**/obj/
-src/**/bin/
-buildscripts/**/obj/
-buildscripts/**/bin/
-project.lock.json
-/.dnx/
 /.dotnet/
-tools/**/obj/
+**/obj/**
+**/bin/**
 tools/**/build/
+
+# Xml doc/test files
+*.xml
```

Comment:
> Mind you, this is only the root folder. It also prevents DesktopClrTestResults.xml and NetCoreClrTestResults.xml from being unstaged and accidentally committed.

@fir3pho3nixx I think you misunderstand how gitignores work, sorry for this pull request being death by a thousand cuts.

> If the pattern does not contain a slash /, Git treats it as a shell glob pattern and checks for a match against the pathname relative to the location of the .gitignore file (relative to the toplevel of the work tree if not from a .gitignore file).
>
> If the pattern ends with a slash, it is removed for the purpose of the following description, but it would only find a match with a directory. In other words, foo/ will match a directory foo and paths underneath it, but will not match a regular file or a symbolic link foo (this is consistent with the way how pathspec works in general in Git).
>
> A leading slash matches the beginning of the pathname. For example, "/*.c" matches "cat-file.c" but not "mozilla-sha1/sha1.c".
>
> (https://git-scm.com/docs/gitignore)

`*.xml` is a glob pattern across all directories (like the `*.suo` pattern), it would need to be `/*.xml` to just match the root.

There is also no need for the `**`, you can just do `bin/` to match the directory rather the contents.

Reply:
Okiday, I checked the globs by running git status and it appeared to be OK. Will fix.

Reply:
Cheers, I know I'm a nitpicking bastard 😉 

P.S. this PR now also conflicts because of another PR I merged earlier.

Reply:
Don't worry I will sort that.

Reply:
@jonorossi - I think I misled you earlier when I said the XML artifacts are only in the root. 

I cleaned the project, using `git clean -x -f -d`, then changed the `.gitignore` to have a leading slash in front of the `*.xml` -> `/*.xml`. 

After running a full build and tests, I then double checked using `git status` with the following result. 

```
Your branch is up-to-date with 'origin/harmonising-core-and-windsor-2'.

All conflicts fixed but you are still merging.
  (use "git commit" to conclude merge)

Changes to be committed:

        modified:   CHANGELOG.md
        modified:   src/Castle.Services.Logging.log4netIntegration/Castle.Services.Logging.log4netIntegration.csproj

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   .gitignore
        modified:   Castle.Core.sln.DotSettings

Untracked files:
  (use "git add <file>..." to include in what will be committed)

        src/Castle.Services.Logging.log4netIntegration/Castle.Services.Logging.Log4netIntegration.xml
``` 

Notice how git thinks it needs to add the xml build artifact from the Log4netIntegration?

I believe my original approach is correct because this is a build artifact. It was my fault for not being clear on this in the first place but I do test my work before committing(admittedly I don't always test it on linux hence the previous casing issue). I am also puzzled as to why the casing issue cropped up and was not detected by Travis. Bizarre. 

The globbing pattern I thought would be OK because of this:

```
Two consecutive asterisks ("**") in patterns matched against full pathname may have special meaning:

A leading "**" followed by a slash means match in all directories. For example, "**/foo" matches file or directory "foo" anywhere, the same as pattern "foo". "**/foo/bar" matches file or directory "bar" anywhere that is directly under directory "foo". <- !! Why I put the glob on the front of bin/obj. !!

A trailing "/**" matches everything inside. For example, "abc/**" matches all files inside directory "abc", relative to the location of the .gitignore file, with infinite depth. <- !! Why I put the glob on the end !!

A slash followed by two consecutive asterisks then a slash matches zero or more directories. For example, "a/**/b" matches "a/b", "a/x/b", "a/x/y/b" and so on.

Other consecutive asterisks are considered invalid.
```

You might consider it unnecessary although it is still valid. I just failed to explain myself perhaps properly. 

I am going to fix the merge conflict and add `Castle.Core.sln.DotSettings` to the gitignore. 

If none of this is good for you, my alternative is to close out the PR. I am done wasting time on this.